hat/assets/js/apps/Iaso/domains/users/components/ProtectedRoute.js

```diff
@@ -1,109 +1,92 @@
-            this.props;
-        if (currentUser && currentUser !== prevProps.currentUser) {
-            const isAuthorized =
-                permissions.length > 0
-                    ? userHasOneOfPermissions(permissions, currentUser)
-                    : true;
-            if (!isAuthorized && isRootUrl) {
-                const newBaseUrl = getFirstAllowedUrl(
-                    permissions,
-                    currentUser?.permissions ?? [],
-                    allRoutes,
-                );
-                if (newBaseUrl) {
-                    redirectTo(newBaseUrl, {});
-                }
-            }
-            // Use defined default language if it exists and if the user didn't set it manually
-            if (
-                currentUser?.language &&
-                currentUser?.language !== prevProps.currentUser?.language
-            ) {
-                this.props.dispatch(switchLocale(currentUser.language));
-            }
+    useEffect(() => {
+        if (!params.accountId && currentUser.account) {
+            dispatch(
+                redirectTo(baseUrl, {
+                    ...params,
+                    accountId: currentUser.account.id,
+                }),
+            );
         }
-    }
+    }, [currentUser.account, baseUrl, params, dispatch]);
+
+    useEffect(() => {
+        // Use defined default language if it exists and if the user didn't set it manually
+        if (currentUser.language) {
+            dispatch(switchLocale(currentUser.language));
+        }
+    }, [currentUser.language, dispatch]);
 
-    render() {
-        const { component, currentUser, permissions, featureFlag, location } =
-            this.props;
-        let isAuthorized =
+    useEffect(() => {
+        const isAuthorized =
             permissions.length > 0
                 ? userHasOneOfPermissions(permissions, currentUser)
                 : true;
-        if (featureFlag && !hasFeatureFlag(currentUser, featureFlag)) {
-            isAuthorized = false;
-        }
-        if (!currentUser) {
-            return null;
+        if (!isAuthorized && isRootUrl) {
+            const newBaseUrl = getFirstAllowedUrl(
+                permissions,
+                currentUser.permissions ?? [],
+                allRoutes,
+            );
+            if (newBaseUrl) {
+                dispatch(redirectTo(newBaseUrl, {}));
+            }
         }
-        return (
-            <>
-                <SidebarMenu location={location} />
-                {isAuthorized && component}
-                {!isAuthorized && <PageError errorCode="401" />}
-            </>
-        );
+    }, [allRoutes, currentUser, dispatch, isRootUrl, permissions]);
+
+    let isAuthorized =
+        permissions.length > 0
+            ? userHasOneOfPermissions(permissions, currentUser)
+            : true;
+    if (featureFlag && !hasFeatureFlag(currentUser, featureFlag)) {
+        isAuthorized = false;
```

Comment:
Why not define that earlier and use it in the useEffect above?
Also, line 65, I'm not sure I'm conmfortable with `isAuthorized`  being et to `true` by default. I guess it might be a byproduct of transposing `componentDidUpdate` to `useEffect`. If so, maybe it can be avoided by using `useSkipEffectOnMount` (I don't know, maybe you tried and it didn't work, I'm just thinking out loud really)

Reply:
Ok, I just saw that we set it to `true` in the old code as well. No idea why though